src/data-store-algorithm/data-store-in-note-algorithm-osr.ts

```diff
@@ -51,30 +52,43 @@ export class DataStoreInNoteAlgorithmOsr implements IDataStoreAlgorithm {
 
         const schedInfo: RepItemScheduleInfoOsr = repItemScheduleInfo as RepItemScheduleInfoOsr;
         const dueString: string = formatDateYYYYMMDD(schedInfo.dueDate);
-        const interval: number = schedInfo.interval;
         const ease: number = schedInfo.latestEase;
+        const interval: number = schedInfo.interval;
 
         // check if scheduling info exists
-        if (SCHEDULING_INFO_REGEX.test(fileText)) {
-            const schedulingInfo = SCHEDULING_INFO_REGEX.exec(fileText);
+        const hasSchedulingDueString = SCHEDULING_INFO_DUE_REGEX.test(fileText);
+        const hasSchedulingEase = SCHEDULING_INFO_EASE_REGEX.test(fileText);
+        const hasSchedulingInterval = SCHEDULING_INFO_INTERVAL_REGEX.test(fileText);
+        if (hasSchedulingDueString && hasSchedulingEase && hasSchedulingInterval) {
+            const schedulingDueString = SCHEDULING_INFO_DUE_REGEX.exec(fileText);
+            fileText = fileText.replace(
+                SCHEDULING_INFO_DUE_REGEX,
+                `${schedulingDueString[1]}${dueString}${schedulingDueString[3]}`,
+            );
+
+            const schedulingEase = SCHEDULING_INFO_EASE_REGEX.exec(fileText);
+            fileText = fileText.replace(
+                SCHEDULING_INFO_EASE_REGEX,
+                `${schedulingEase[1]}${ease}${schedulingEase[3]}`,
+            );
+
+            const schedulingInterval = SCHEDULING_INFO_INTERVAL_REGEX.exec(fileText);
             fileText = fileText.replace(
```

Comment:
I've just checked the Obsidian API and it's now possible to modify the FrontMatter using the native `processFrontMatter` function. That might be the better approach.

https://github.com/obsidianmd/obsidian-api/blob/23947b58d372ea02225324308e31d36b4aa95869/obsidian.d.ts#L1551

>   /**
     * Atomically read, modify, and save the frontmatter of a note.
     * The frontmatter is passed in as a JS object, and should be mutated directly to achieve the desired result.
     *
     * Remember to handle errors thrown by this method.
     *
     * @param file - the file to be modified. Must be a Markdown file.
     * @param fn - a callback function which mutates the frontmatter object synchronously.
     * @param options - write options.
     * @throws YAMLParseError if the YAML parsing fails
     * @throws any errors that your callback function throws
     * @example
     * ```ts
     * app.fileManager.processFrontMatter(file, (frontmatter) => {
     *     frontmatter['key1'] = value;
     *     delete frontmatter['key2'];
     * });
     * ```
     * @public
     */
    processFrontMatter(file: TFile, fn: (frontmatter: any) => void, options?: DataWriteOptions): Promise<void>;